### DISCUSSION TITLE
Chapter 01. 처음 만나는 쿠버네티스 - 02. 쿠버네티스와 클러스터 동작 방식 강의 자료 - 16번 Slide 질문

### DISCUSSION THREAD

**Discussion by cypision** [yoonjeong-kwon/fastcampus-kubernetes#21] [Q&A] [answered]
안녕하세요. 
본 장표에서 요청 받은 "10.80.13.74" 로 packet 를 전달할때, 임의의 컨테이너로 리다이렉트 된다고 설명해주셨는데요, 
실제 가야 하는(옳바른) Conatainer 로 가는게 최초 요청자의 목적일텐데....
혹 설명해주신 것 처럼, iptables에 있는 임의의 컨테이너로, 리다이렉트 되었을 때, 해당 Container가 응답이 없으면, 바로 다른 Container로 또 Redirect 되는 건가요? 
 
 장표의 10.70.0.231:8080 POD와 10.76.1.53:8080 POD 는 같은 역할을 하는 POD 인가요? 마치 Replication 2개 띄운 것처럼?

**Comment by yoonjeong-kwon** [comment, marked as answer]
@cypision 님, 안녕하세요. 
iptables 방식에서는 임의의 컨테이너로 요청이 전달되었을 때
컨테이너에서 응답이 없으면 요청이 실패하고 재시도 하지 않습니다.
그리고 iptables를 구성할 때 파드(컨테이너)가 정상적으로 응답할 수 있는지를 readiness probe로 검사하고 리스트를 구성합니다.
kube-proxy에 대한 그림과 추가 설명을 보실 수 있도록 공식 문서 공유드립니다.
https://kubernetes.io/ko/docs/concepts/services-networking/service/#proxy-mode-iptables

질문주신 10.70.0.231:8080 POD와 10.76.1.53:8080 POD 가 어느 장표에 나와 있는지 정확히 찾지 못했습니다.
장표에 파드 이름이 있다면 파드 이름을 알려 주실 수 있을까요? 아니면 스냅샷 첨부 부탁드리겠습니다.
감사합니다!

**Reply by cypision** [reply]
"kube-proxy와 iptable을 이용한 통신 방법" 에서, 그림에 나와있는 POD입니다.  iptables 설명할 때 보면, 밑에 Docker-POD들의 주소가
10.70.0.231:8080 POD와 10.76.1.53:8080 POD 입니다.
질문을 구체화하면, 상기 장표에서 
iptables 방식에서는 최종 목적지 10.80.13.74 에 도달할때, 10.70.0.231:8080 POD와 10.76.1.53:8080 POD 중 1개만 임의로 선택해서 보낸다고 하셨는데, 10.70.0.231:8080 POD와 10.76.1.53:8080 POD 는 같은 서비스인가요? 그러니깐, Replica=2 로 띄운거라고 생각하면 되나요?

**Reply by yoonjeong-kwon** [reply]
@cypision 님, 질문 확인이 많이 늦었습니다. 죄송합니다.
이해하신 바가 맞습니다 :)
Pod가 2개 떠있는, 즉 Replica가 2로 선언되어 있는 것이 맞습니다.
